### PR TITLE
Implement threshold decision policy

### LINKED ISSUE #45: Implement ThresholdDecisionPolicy
needed-by #31

### PR DESCRIPTION
**Pull request by alpe**
Resolves #45

After #54

### CODE REVIEW

File: incubator/group/types.go
Comment (aaronc) [inline incubator/group/types.go]:
```suggestion
// DecisionPolicy is persistent set of rules to determine the result of election on a proposal.
```

File: incubator/group/types.go
Comment (aaronc) [inline incubator/group/types.go]:
```suggestion
// Allow allows a proposal to pass when the tally of yes votes equals or exceeds the threshold before the timeout.
```

File: incubator/group/types.proto
Comment (aaronc) [inline incubator/group/types.proto]:
```suggestion
    // threshold is a fix weight value that must be met or exceeded for a proposal to succeed.
```

File: incubator/group/types_test.go
Comment (aaronc) [inline incubator/group/types_test.go]:
```suggestion
		"accept when yes count equal to threshold": {
```

File: incubator/group/types_test.go
Comment (aaronc) [inline incubator/group/types_test.go]:
```suggestion
			expResult:         DecisionPolicyResult{Allow: true, Final: true},
```

File: incubator/group/types_test.go
Comment (aaronc) [inline incubator/group/types_test.go:52]: Update according to new definition of threshold

File: incubator/group/types_test.go
Comment (aaronc) [inline incubator/group/types_test.go]:
```suggestion
		"abstain has no impact": {
```

**Review by aaronc — CHANGES_REQUESTED** [review]
The tests look good but I realize that the threshold policy is actually incorrectly specified. The threshold must be _met or_ exceeded. Just exceeding it is counter intuitive because in order to have a 1 out of 3 policy, you would need to set the threshold to 0. Looks like I messed up with my initial specification. My apologies!

**Comment by alpe** [comment]
I have modified the policy. With equal numbers we need to prohibit threshold==0, now so that we have a least 1 yes vote.

File: incubator/group/types.go
Comment (aaronc) [inline incubator/group/types.go]:
```suggestion
// DecisionPolicy is the persistent set of rules to determine the result of election on a proposal.
```

**Review by aaronc — CHANGES_REQUESTED** [review]
Just found another minor doc type. Otherwise LGTM and ready to merge.

**Comment by alpe** [comment]
I had applied the change. Waiting for your approval (on github)

**Review by aaronc — APPROVED** [review]

**Pull request merged**
